actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -671,12 +675,18 @@ def use_relative_controller!
 
         # Remove leading slashes from controllers
         def normalize_controller!
-          @options[:controller] = controller.sub(%r{^/}, ''.freeze) if controller
+          if controller
+            if m = controller.match(/\A\/(?<controller_without_leading_slash>.*)/)
+              @options[:controller] = m[:controller_without_leading_slash]
+            else
+              @options[:controller] = controller
+            end
+          end
         end
 
         # Move 'index' action from options to recall
         def normalize_action!
-          if @options[:action] == 'index'
+          if @options[:action] == 'index'.freeze
```

Comment:
Check out https://github.com/rails/rails/pull/21057/files#r36532115 for why we're not doing that